sklearn/neighbors/_classification.py

```diff
@@ -57,42 +58,50 @@ class KNeighborsClassifier(NeighborsBase, KNeighborsMixin,
         Note: fitting on sparse input will override the setting of
         this parameter, using brute force.
 
-    leaf_size : int, optional (default = 30)
+    leaf_size : int, default=30
         Leaf size passed to BallTree or KDTree.  This can affect the
         speed of the construction and query, as well as the memory
         required to store the tree.  The optimal value depends on the
         nature of the problem.
 
-    p : integer, optional (default = 2)
+    p : int, default=2
         Power parameter for the Minkowski metric. When p = 1, this is
         equivalent to using manhattan_distance (l1), and euclidean_distance
         (l2) for p = 2. For arbitrary p, minkowski_distance (l_p) is used.
 
-    metric : string or callable, default 'minkowski'
-        the distance metric to use for the tree.  The default metric is
+    metric : {'cityblock', 'cosine', 'euclidean', 'l1', 'l2', 'manhattan',
```

Comment:
Sorry, I went a bit fast. There is too much metrics to be enumerated. could you revert to `str` instead.